packages/flutter_tools/lib/src/flutter_manifest.dart

```diff
@@ -181,29 +172,29 @@ class FlutterManifest {
   /// Returns the Android package declared by this manifest in its
   /// module or plugin descriptor. Returns null, if there is no
   /// such declaration.
-  String get androidPackage {
+  String? get androidPackage {
     if (isModule) {
-      return _flutterDescriptor['module']['androidPackage'] as String;
+      return _flutterDescriptor['module']['androidPackage'] as String?;
     }
     if (supportedPlatforms == null) {
       // Pre-multi-platform plugin format
       if (isPlugin) {
-        final YamlMap plugin = _flutterDescriptor['plugin'] as YamlMap;
-        return plugin['androidPackage'] as String;
+        final YamlMap? plugin = _flutterDescriptor['plugin'] as YamlMap?;
+        return plugin?['androidPackage'] as String?;
       }
       return null;
     }
-    if (supportedPlatforms.containsKey('android')) {
-       return supportedPlatforms['android']['package'] as String;
+    if (supportedPlatforms!.containsKey('android')) {
+       return supportedPlatforms!['android']['package'] as String?;
     }
     return null;
   }
 
   /// Returns the deferred components configuration if declared. Returns
   /// null if no deferred components are declared.
-  List<DeferredComponent> get deferredComponents => _deferredComponents ??= computeDeferredComponents();
-  List<DeferredComponent> _deferredComponents;
-  List<DeferredComponent> computeDeferredComponents() {
+  List<DeferredComponent>? get deferredComponents => _deferredComponents ??= computeDeferredComponents();
```

Comment:
This is probably fine, but its a little weird to do the `_deferredComponents ??= ` trick on a field that could be nullable. Instead, you could make this `late final` (which lacks many of the problems of regular late:

```
late final deferredComponents  = computeDeferredComponents();
```

Then `computeDeferredComponents` will only ever run once, even if it is null

Reply:
Oh right, `late final`.  Good call.